python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -515,6 +518,147 @@ def get_dynamic_partitions(self, partitions_def_name: str) -> Sequence[str]:
     def has_dynamic_partition(self, partitions_def_name: str, partition_key: str) -> bool:
         return partition_key in self.get_dynamic_partitions(partitions_def_name)
 
+    @cached_method
+    def asset_partitions_with_newly_updated_parents_and_new_latest_storage_id(
+        self,
+        *,
+        latest_storage_id: Optional[int],
+        target_asset_keys: FrozenSet[AssetKey],
+        target_asset_keys_and_parents: FrozenSet[AssetKey],
+        can_reconcile_fn: Callable[[AssetKeyPartitionKey], bool] = lambda _: True,
```

Comment:
Will this go away in some way in the future? A little weird to be caching on this.

Reply:
yeah this can be removed at any time (just didn't want to comingle that mild implementation change with this)